src/main/java/core/basesyntax/strategy/impl/PurchaseTransaction.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.strategy.FruitTransaction;
+
+public class PurchaseTransaction implements FruitTransaction {
+    private StorageDao storageDao = new StorageDaoImpl();
+
+    @Override
+    public void implementTransactionWithFruit(String fruit, int quantity) {
+        storageDao.putInStorage(fruit, (storageDao.getQuantity(fruit) - quantity));

```

Comment:
```suggestion
        storageDao.updateData(fruit, (storageDao.getQuantity(fruit) - quantity));

```
You are updating already existed data, so it should look like that